src/config/PluginConfigManager.ts

```diff
@@ -110,7 +111,7 @@ export class PluginConfigManager<TPluginType extends BasePluginType> {
     const level = matchParams?.level ?? (member && this.getMemberLevel(member)) ?? null;
 
     // Passed roles -> passed member's roles
-    const memberRoles = matchParams.memberRoles || (member && member.roles.cache.keyArray());
+    const memberRoles = matchParams.memberRoles ?? (member && member.roles?.cache.keyArray()) ?? [];
```

Comment:
Is there a situation where a member doesn't have the roles object? Partials?

Reply:
No. `GuildMember#roles` is always present because it's a manager

Reply:
The event was sometimes fired twice, the second time probably originating from some other voice event.
This is a band-aid fix on purpose, since this needs deeper investigation.